keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -30,9 +31,26 @@
     private final int TOP_LEVEL_INDENTATION = 1;
     private int indentation = TOP_LEVEL_INDENTATION;
 
+    private final int vertexCount;
+    private final Map<VertexId, Integer> vertexDegrees;
+    private final double averageVertexDegree;
+
     public BayesianNetwork(Set<? extends Vertex> vertices) {
         this.vertices = ImmutableList.copyOf(vertices);
         this.vertexLabels = buildLabelMap(vertices);
+
+        this.vertexCount = vertices.size();
+        this.vertexDegrees = calculateVertexDegrees(vertices);
+        this.averageVertexDegree = calculateAverageVertexDegree(vertices);
```

Comment:
This class is already quite busy - I would prefer to have any metrics-related functionality in a separate class.